addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+                            + ")."));
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+            logger.trace("initialize(): finished with status ONLINE.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        logger.trace("dispose() called.");
+        refreshJob.cancel(true);
+        super.dispose();
+    }
+
+    /**
+     * Start the periodic information retrieval from the Solarview server.
+     */
+    private void startAutomaticRefresh() {
+        logger.trace("startAutomaticRefresh() called.");
+        refreshJob = scheduler.scheduleAtFixedRate(() -> {
+            try {
+                boolean success = updateEnergyDataFromServer();
+                if (success) {
+                    for (int i = 0; i < channels.length; i++) {
+                        updateState(new ChannelUID(getThing().getUID(), channels[i]), getChannelValue(channels[i]));
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }, 0, refreshInterval.intValue(), TimeUnit.SECONDS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand({},{}) called.",channelUID.getAsString(),command.toString());
+
+        if (command instanceof RefreshType) {
+            boolean success = updateEnergyDataFromServer();
+            if (success) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_KYR:
+                    case CHANNEL_KMT:
+                    case CHANNEL_KDY:
+                    case CHANNEL_KT0:
+                    case CHANNEL_PAC:
+                    case CHANNEL_UDC:
+                    case CHANNEL_IDC:
+                    case CHANNEL_UDCB:
+                    case CHANNEL_IDCB:
+                    case CHANNEL_UDCC:
+                    case CHANNEL_IDCC:
+                    case CHANNEL_UL1:
+                    case CHANNEL_IL1:
+                    case CHANNEL_TKK:
+                        updateState(channelUID, getChannelValue(channelUID.getId()));
+                        break;
+                    default:
+                        logger.error("Command received for an unknown channel: {}", channelUID.getId());
+                        break;
+                }
+            }
+        } else {
+            logger.error("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
```

Comment:
No, you're right: reverting to `BaseThingHandler` eliminates the need for this cruel method.

Fixed.